src/main/java/org/projectnessie/tool/CatalogMigrateUtil.java

```diff
@@ -0,0 +1,178 @@
+   *     migrated. If not specified, all the tables would be migrated
+   * @param sourceCatalog Source {@link Catalog} from which the tables are chosen
+   * @param targetCatalog Target {@link Catalog} to which the tables need to be migrated
+   * @param maxThreadPoolSize Size of the thread pool used for migrate tables (If set to 0, no
+   *     thread pool is used)
+   * @return Collection of table identifiers for successfully migrated tables
+   */
+  public static Collection<TableIdentifier> migrateTables(
+      List<TableIdentifier> tableIdentifiers,
+      Catalog sourceCatalog,
+      Catalog targetCatalog,
+      int maxThreadPoolSize) {
+    return migrateTables(tableIdentifiers, sourceCatalog, targetCatalog, maxThreadPoolSize, true);
+  }
+
+  /**
+   * Register tables from one catalog(source catalog) to another catalog(target catalog). User has
+   * to take care of deleting the tables from source catalog after registration.
+   *
+   * <p>Supports bulk registration with a multi-thread execution.
+   *
+   * <p>Users must make sure that no in-progress commits on the tables of source catalog during
+   * registration.
+   *
+   * @param tableIdentifiers a list of {@link TableIdentifier} for the tables required to be
+   *     registered. If not specified, all the tables would be registered
+   * @param sourceCatalog Source {@link Catalog} from which the tables are chosen
+   * @param targetCatalog Target {@link Catalog} to which the tables need to be registered
+   * @param maxThreadPoolSize Size of the thread pool used for registering tables (If set to 0, no
+   *     thread pool is used)
+   * @return Collection of table identifiers for successfully registered tables
+   */
+  public static Collection<TableIdentifier> registerTables(
+      List<TableIdentifier> tableIdentifiers,
+      Catalog sourceCatalog,
+      Catalog targetCatalog,
+      int maxThreadPoolSize) {
+    return migrateTables(tableIdentifiers, sourceCatalog, targetCatalog, maxThreadPoolSize, false);
+  }
+
+  private static Collection<TableIdentifier> migrateTables(
+      List<TableIdentifier> tableIdentifiers,
+      Catalog sourceCatalog,
+      Catalog targetCatalog,
+      int maxThreadPoolSize,
+      boolean deleteEntriesFromSourceCatalog) {
+    validate(sourceCatalog, targetCatalog, maxThreadPoolSize);
+
+    List<TableIdentifier> identifiers;
+    if (tableIdentifiers == null || tableIdentifiers.isEmpty()) {
+      // fetch all the table identifiers from all the namespaces.
+      List<Namespace> namespaces =
+          (sourceCatalog instanceof SupportsNamespaces)
+              ? ((SupportsNamespaces) sourceCatalog).listNamespaces()
+              : ImmutableList.of(Namespace.empty());
+      identifiers =
+          namespaces.stream()
+              .flatMap(namespace -> sourceCatalog.listTables(namespace).stream())
+              .collect(Collectors.toList());
```

Comment:
This is a potentially expensive/long-running operation, right. I'd prefer to split this one out and give users feedback about what the tool is doing. Listing tables from multiple namespaces is likely something that _can_ be parallelized.